src/components/table/DataTable.tsx

```diff
@@ -1,157 +1,98 @@
-        <Button binIcon={true} />
-      </>
-    ),
-  },
-  {
-    name: "Ruben Werdmulier Von Elg",
-    data: "DEGO DOOK Autobranche",
-    mail: "ruben@ontwikkelaar.nl",
-    verified: true,
-    account: "2 nov 2022",
-    active: "2 nov 2022 11:05",
-    action: (
-      <>
-        <Button writeIcon={true} />
-        <Button lockCloseIcon={true} />
-        <Button binIcon={true} />
-      </>
-    ),
-  },
-  {
-    name: "Stephan de Preeker",
-    data: "-",
-    mail: "stephan@memory.com",
-    verified: false,
-    account: "2 nov 2022",
-    active: "2 nov 2022 11:05",
-    action: (
-      <>
-        <Button writeIcon={true} />
-        <Button lockCloseIcon={true} />
-        <Button binIcon={true} />
-      </>
-    ),
-  },
-];
+import { MenuItem, InputAdornment, Table, TableRow } from "@mui/material";
+import { Button } from "../button/Button";
 
 export const DataTable = () => {
   const [sortOrder, setSortOrder] = useState("");
   const [sortedTable, setSortedTable] = useState(data);
   const [searchValue, setSearchValue] = useState("");
 
-  const sortOptions = ["Naam", "E-mail", "Geverifieerd"];
+  const sortOptions = [
+    {
+      label: "fullName",
+      value: "Name",
+    },
+    {
+      label: "mail",
+      value: "E-mail",
+    },
+    {
+      label: "verified",
+      value: "Geverifieerd",
+    },
+    {
+      label: "accesTo",
+      value: "Toegang",
+    },
+  ];
+
+  const sortKey = sortOptions.map((entry) => entry.label);
+  const buttonIcons = data.map((entry) => entry.setActions);
+
+  const filteredTable = sortedTable.filter((value) => {
+    if (searchValue === "") {
+      return value;
+    } else if (
+      value.fullName.toLowerCase().includes(searchValue.toLocaleLowerCase()) ||
+      value.mail.toLowerCase().includes(searchValue.toLocaleLowerCase())
+    ) {
+      return value;
+    }
+  });
 
-  const handleSort = () => {
+  const handleSort = (sortKey: string) => {
     const sort = data.sort((a, b) => {
-      if (a.name < b.name) {
-        return sortOrder === "Naam" ? 1 : -1;
-      }
-      if (b.mail < a.mail) {
-        return sortOrder === "E-mail" ? 1 : -1;
+      const keyA = a[sortKey];
+      const keyB = b[sortKey];
+
+      console.log(sortKey);
+
+      if (keyA < keyB) {
+        return -1;
       }
-      if (a.verified < b.verified) {
-        return sortOrder === "Geverifieerd" ? 1 : -1;
+      if (keyA > keyB) {
+        return 1;
       }
-
       return 0;
     });
 
     setSortedTable(sort);
   };
 
   useEffect(() => {
-    handleSort();
+    handleSort(sortKey as string | any);
   }, [sortOrder, data]);
 
   return (
     <>
       <Styled.Wrapper>
         <div>
-          <Styled.Label htmlFor="">Sorteer op</Styled.Label>
-          <Styled.StyledSelect
+          <Styled.Label>Sorteer op</Styled.Label>
+          <Styled.CustomSelect
             value={sortOrder}
             onChange={(e) => setSortOrder(e.target.value as string)}
           >
```

Comment:
Since you now handle your sort based on the clicking of menu items, do you still need this function?